squad/core/history.py

```diff
@@ -28,39 +35,56 @@ class TestHistory(object):
 
     __test__ = False
 
+    def __get_tests__(self, builds, metadata):
+        test_runs = []
+        for build in builds:
+            for test_run in build.test_runs.all():
+                test_runs.append(test_run)
+
+        tests = []
+        for test_runs_batch in split_list(test_runs, chunk_size=100):
```

Comment:
maybe it would be even faster to do sth like this:
```
test_run_ids = []
for build in builds:
    test_run_ids += build.test_runs.all().values_list('id', flat=True)

tests = Test.objects.filter(test_run__id__in=test_run_ids)
```
I didn't test this code

Reply:
That's exactly what's happening in `prefetch_related_objects(test_runs_batch, Prefetch('tests',`, it translates to something like:

```sql
SELECT "core_test"."id", ...
  FROM "core_test"
 WHERE ("core_test"."metadata_id" = 149002 AND "core_test"."test_run_id" IN (955445, 955501, 955813, ...))
```